src/bin/sol-fbp-generator/main.c

```diff
@@ -633,8 +658,12 @@ generate_node_type_assignments(const struct fbp_data *data)
     out("\n");
 
     SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
-        struct type_description *desc = n->user_data;
-        out("    nodes[%d].type = %s;\n", i, desc->symbol);
+        struct node_data *nd = n->user_data;
+
+        if (nd->is_declared)
```

Comment:
On Sep 24, 2015, at 18:14, Gustavo Sverzut Barbieri <notifications@github.com<mailto:notifications@github.com>> wrote:


In src/bin/sol-fbp-generator/main.c<https://github.com/solettaproject/soletta/pull/782#discussion_r40374136>:

> @@ -633,8 +658,12 @@ generate_node_type_assignments(const struct fbp_data *data)
>      out("\n");
>
>      SOL_VECTOR_FOREACH_IDX (&data->graph.nodes, n, i) {
> -        struct type_description *desc = n->user_data;
> -        out("    nodes[%d].type = %s;\n", i, desc->symbol);
> +        struct node_data *nd = n->user_data;
> +
> +        if (nd->is_declared)


hum... maybe check nd->desc or nd->desc->symbol and remove nd->is_declared? is it possible?

Both exist, I'd need to check for a given pattern or something. I don't see why removing is_declared would gain much.

�
Reply to this email directly or view it on GitHub<https://github.com/solettaproject/soletta/pull/782/files#r40374136>.